tests/php/ORM/DataQueryTest.php

```diff
@@ -879,4 +879,48 @@ public function testWithUsingDataQueryAppliesRelations()
         // This will throw an exception if it fails - it passes if there's no exception.
         $dataQuery->execute();
     }
+
+    public function testDistinctCount()
+    {
+        // The COUNT(DISTINCT ...) is added in case a join is added to the query (to apply a filter on related records)
+        $distinct = new DataQueryTest\ObjectC();
+        $distinct->Title = 'Distinct';
+        $distinct->write();
+
+        $otherDistinct = new DataQueryTest\ObjectC();
+        $otherDistinct->Title = 'OtherDistinct';
+        $otherDistinct->write();
+
+        $baseCount = DataQueryTest\ObjectA::get()->count();
+
+        $obj1 = new DataQueryTest\ObjectA();
+        $obj1->Name = 'sam';
+        $obj1->TestCID = $distinct->ID;
+        $obj1->write();
+
+        $obj2 = new DataQueryTest\ObjectA();
+        $obj2->Name = 'sam';
+        $obj2->TestCID = $distinct->ID;
+        $obj2->write();
+
+        $obj3 = new DataQueryTest\ObjectA();
+        $obj3->Name = 'john';
+        $obj3->TestCID = $otherDistinct->ID;
+        $obj3->write();
```

Comment:
Can you please set up fixtures for these? Ideally the test method should just include the logic required to perform the test itself.